scripts/update-frontend-canister.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -e
+die() {
+  echo "$1" >&2
+  exit 1
+}
+
+cargo --version >/dev/null || die "Must have cargo installed."
+
+export RUSTFLAGS="--remap-path-prefix=\"${PWD}\"=./ --remap-path-prefix=\"${HOME}\"=_/"
+cargo build -p certified-assets --release --target wasm32-unknown-unknown
+
+BUILD_DIR="target/wasm32-unknown-unknown/release"
+ic-wasm --output $BUILD_DIR/certified_assets.wasm $BUILD_DIR/certified_assets.wasm metadata --file src/canisters/frontend/ic-certified-assets/assets.did --visibility public candid:service
+ic-wasm --output $BUILD_DIR/certified_assets.wasm $BUILD_DIR/certified_assets.wasm shrink
+gzip --best --keep --force --no-name $BUILD_DIR/certified_assets.wasm
+
+cp -f $BUILD_DIR/certified_assets.wasm.gz src/distributed/assetstorage.wasm.gz
```

Comment:
@adamspofford-dfinity @ericswanson-dfinity 
I'm gzipping asset canister wasm here. I changed the extension of `src/distributed/assetstorage.` from `wasm` to `wasm.gz`. That unfortunately doesn't fly in our CI: https://github.com/dfinity/sdk/runs/7992839597?check_suite_focus=true#step:8:1053, the issue is that our canister installer now expects to find candid file in 
`src/distributed/assetstorage.wasm.did`. 

My initial guess is we're using Rust's `.with_extension` method on path, which replaces text after last dot, here is what I found: https://sourcegraph.com/github.com/dfinity/sdk/-/blob/src/dfx/src/lib/builders/assets.rs?L194 (I haven't worked with dfx code before, so it may be a miss).

Obviously two quick fixes are either
```
mv src/distributed/assetstorage.did src/distributed/assetstorage.wasm.did
```
or go back to
```
mv src/distributed/assetstorage.wasm.gz src/distributed/assetstorage.wasm 
```

Let me know if you think I should use either of those quick-fixes or should I wait for the fix upstream?

Reply:
The filenames should be `assetstorage.wasm.gz` and `assetstorage.did`.

You may need to do something like `path.with_extension("").with_extension("did")`, but in any case, that is a solvable problem.

Reply:
I guess what I really wanted to know is whether I found the correct LoC in the dfx source which causes it, and, whether I should bring the change, or maybe it's already on the radar of either one of you guys.

Reply:
btw fyi
```rust
    let p = std::path::Path::new("/wow");
    let a = p.join("thing.wasm.mm");
    dbg!(&a);
    dbg!(&a.with_extension("").with_extension(".did"));
    
    let p = std::path::Path::new("/wow");
    let a = p.join("thing.wasm");
    dbg!(&a);
    dbg!(&a.with_extension("").with_extension(".did"));
```
prints 
```
[src/main.rs:16] &a = "/wow/thing.wasm.mm"
[src/main.rs:17] &a.with_extension("").with_extension(".did") = "/wow/thing..did"
[src/main.rs:21] &a = "/wow/thing.wasm"
[src/main.rs:22] &a.with_extension("").with_extension(".did") = "/wow/thing..did"
```

@adamspofford-dfinity if you know any party trick like `.with_all_extensions` or something, let me know :)

Reply:
Try without a `.` in the with_extension parameter

```
path.with_extension("").with_extension("did")
```

Reply:
Wow, I've simply outdone myself here :D
Thanks!